File: pset6/vigenere/vigenere.py
# Vigenere's algorithm

import sys
from cs50 import get_string

# Check if there exist two arguments
if len(sys.argv) != 2:
    print("You have to pass TWO arguments!")
    sys.exit(1)

# Get the key
key = sys.argv[1]
for c in key:
    if not c.isalpha():
        print("There is non-alphabetical character in the key!")
        sys.exit(1)

# Prompt user to enter plain text
plainText = get_string("plaintest: ")
chiperText = ""

# Iterate over each character in plain text
j = 0
for i in range(len(plainText)):
    char = plainText[i]

    if char.isalpha():
        j %= len(key)
        # Convert key's i's character to it's alphabetical index
        k = (ord(key[j]) - 65) if key[j].isupper() else (ord(key[j]) - 97)

        if char.islower():
            # ord("a") equals 97
            char = chr((((ord(plainText[i]) - 97) + k) % 26) + 97)
        else:
            # ord("A") equals 65
            char = chr((((ord(plainText[i]) - 65) + k) % 26) + 65)
        j += 1

    chiperText += char

print(f"ciphertext: {chiperText}")

File: pset2/vigenere/vigenere.c
//Vigenere's algorithm
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("You have to pass TWO arguments!\n");
        return 1;
    }

    string key = argv[1];
    for (int i = 0, n = strlen(key); i < n; i++)
    {
        if (!isalpha(key[i]))
        {
            printf("There is non-alphabetical character in the key!\n");
            return 1;
        }
    }

    string plainText = get_string("plaintext: ");
    for (int i = 0, j = 0, n = strlen(plainText); i < n; i++)
    {
        //check if i'th character is alphabetic
        if (isalpha(plainText[i]))
        {
            j %= strlen(key);
            //convert key's i's character to it's alphabetical index
            int k = isupper(key[j]) ? key[j] - 'A' : key[j] - 'a';

            if (isupper(plainText[i]))
            {
                //convert character to it's alphabetical index
                //and at the finish convert in to it's ASCII code
                plainText[i] = (((plainText[i] - 'A') + k) % 26) + 'A';
            }
            else if (islower(plainText[i]))
            {
                plainText[i] = (((plainText[i] - 'a') + k) % 26) + 'a';
            }
            j++;
        }
    }
    //plainText now converted to chiperText
    printf("ciphertext: %s\n", plainText);
}
